src/ploomber_engine/ipython.py

```diff
@@ -650,26 +655,38 @@ def _execute(self):
 
 
 class IO(StringIO):
-    def __init__(self, initial_value="", newline="\n"):
+    def __init__(self, default, std_type, display=True, newline="\n", initial_value=""):
         super().__init__(initial_value=initial_value, newline=newline)
+        self.default = default
+        self.std_type = std_type
+        self.display = display
         self._values = []
 
     def write(self, s):
         self._values.append(s)
+
+        # Temporary Fix
+        if self.display:
+            if self.std_type == "out":
+                if s != "\n":
+                    self.default.write("\n" + s + "\n")
```

Comment:
Whenever `print` is called in `tqdm`, the `stderr` doesn't add a `\n` automatically, so a leading `\n` is due to that.
If I add `\n` to `stderr`, tqdm will not update in-place and the output will be like papermill

Reply:
Yeah, you are right. Sorry, I was not clear above. 
> Without `tqdm`, everything works fine

Here, I am talking about the papermill like output 

So, to summarize this thread:

1. If I pass stderr and stdout output directly without any modifications:
```python
if self.display:
    if self.std_type == "out":
        self.default.write(s)
    else:
        self.default.write(s)
```
we get this: 
![Raw](https://user-images.githubusercontent.com/43580047/226069606-a89b11a1-946f-4b8f-b37e-0177cca855cf.png)
This is because there is a lack of `\n` in the for loop of `tqdm` and an extra `\r` when `tqdm` terminated

2. If I use `\n` with stderr
```python
if self.display:
    if self.std_type == "out":
        self.default.write(s)
    else:
        self.default.write(s+"\n")
```
![Screenshot_2023-03-20_16-35-19](https://user-images.githubusercontent.com/43580047/226460030-2b8bd602-9299-4a0e-a27d-e81130ec0b5b.png)
![Screenshot_2023-03-20_16-36-37](https://user-images.githubusercontent.com/43580047/226460221-d1a478d7-3a21-4c39-b923-ed5056dab720.png)
`tqdm` is not updated in place because of `\n`, but the `stdout` stream is correct.

3.  Using both leading and ending `\n` in stdout as done above
![](https://user-images.githubusercontent.com/43580047/226115554-8881b4eb-061d-46ea-be46-bc76ede3861e.png)
Extra `\n`, but correct output, tqdm is updated in place,  no character is overwritten

4. Using only leading `\n` in stdout
```python
if self.display:
  if self.std_type == "out":
    if s != "\n":
      self.default.write("\n" + s )
  else:
    self.default.write(s)
```
![Screenshot_2023-03-20_16-46-01](https://user-images.githubusercontent.com/43580047/226462205-a29cbadf-3020-4850-a1cb-dea86adad41b.png)
`0,1,2,..` are missing due to `\r` by `tqdm`. `Hello World 2` is missing due to `\r` by the progress bar of `PloomberEngine`. if I run with `progressbar=False`, we can see `Hello World 2` in the output

PS: A sidenote, the output notebook is littered by tqdm regardless of the approach used: 
![Screenshot_2023-03-20_16-51-03](https://user-images.githubusercontent.com/43580047/226462989-36259ddc-25ea-4b16-878b-fc0841375af2.png)